src/BuildTemplate.ts

```diff
@@ -137,7 +145,13 @@ export default class BuildTemplate {
         }
 
         if (Utils.IsEqual(templateJson.properties.distribute[0].type, "ManagedImage")) {
-            templateJson.properties.distribute[0].imageId = this._taskParameters.imageIdForDistribute;
+            if (this._taskParameters.imageIdForDistribute == "" || this._taskParameters.imageIdForDistribute == undefined) {
```

Comment:
I would rather have this handled in TaskParamters.ts.
All the input validations and manipulations are happening in that class.

Reply:
we dont have subscriptionid in taskparamteres since instance of taskparametrs is created in constructor of imagebuilder and subscriptionid info is fecthed in execute method of imagebuilder.
subscriptionid is used to create managed image